gui.js

```diff
@@ -1552,9 +1552,20 @@ IDE_Morph.prototype.fixLayout = function (situation) {
 IDE_Morph.prototype.setProjectName = function (string) {
     this.projectName = string.replace(/['"]/g, ''); // filter quotation marks
     this.hasChangedMedia = true;
+    this.updateDocumentTitleText();
```

Comment:
Off the top of my head, this is the list. Note, I'm thinking of user actions, not code paths, so some of these might actually call the same functions already:

* We need to be sure we _don't_ switch the title when changing languages of untitled projects.
* opening a project
* saving a new project for the first time
* exporting a new project
* Save As